### PR TITLE
Events reorganization

### PR DESCRIPTION
**Pull request by Kicer86**
This changes reorganize events management.

Idea: instead of one big container for events in WorldController, a stack of IEventLayers is introduced.
Instance of IEventLayer can accept and process event.
Each event is passed to IEventLayer being on a top of IEventLayer stack. If IEventLayer is not interested in event, next IEventLayer from stack gets it.

This allows us to dynamically manage events receivers in easy way.

### CODE REVIEW

**Comment by Kicer86** [comment]
It is a prerequisite for placing buildings task as we will need a temporary (for placing time) layer controlling events.

File: world/WorldController.hpp [hunk 1]
```diff
@@ -8,28 +8,59 @@
...
+     * @brief process given event
+     * @param event event to be processed
+     */
+    virtual void processEvent(const sf::Event& event) const = 0;
+};
 
+
+/**
+ * \brief Class responsible for managing events.
+ */
```
Comment (ThudPoland) [inline world/WorldController.hpp]: Please divide this into proper files.

**Comment by Kicer86** [comment]
> Is this ok to add EventLayers for classes in same source files where classes are? They could be big in future.

They could indeed. I'd propose to move them when needed. Good?

File: main.cpp [hunk 2]
```diff
@@ -19,6 +20,32 @@
 #endif
 #include <cmath>
 
+class WindowResizeEventLayer: public IEventsLayer
```
Comment (Xavrax) [inline main.cpp]: Class in main.cpp is good practice?

Comment (Kicer86) [reply main.cpp]: No opinion about it.

**Pull request merged**